api/planfile_pricer.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"github.com/zparnold/azure-terraform-cost-estimator/common"
+	"golang.org/x/net/context"
+	"sync"
+)
+
+var lock = &sync.Mutex{}
+
+type PlanfilePricer struct {
+	assetPricers         map[string]*common.TerraformPriceableAsset
+	ctx                  context.Context
+	unsupportedResources []string
+}
+
+var singleton *PlanfilePricer
+
+func PlanfilePricerInstance(ctx context.Context) *PlanfilePricer {
+	if singleton == nil {
+		lock.Lock()
+		defer lock.Unlock()
+		if singleton == nil {
```

Comment:
Why are we checking if the singleton is nil twice? Does this capture a race condition?